src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1908,6 +1908,9 @@ public object GetSocketOption(SocketOptionLevel optionLevel, SocketOptionName op
 
             int optionValue = 0;
 
+            if (IOControlKeepAlive.TryGet(this, optionLevel, optionName, ref optionValue))
+                return optionValue;
```

Comment:
It would be better to move this into SocketPal's GetSockOpt implementation, and then only compile IOControlKeepAlive into the Windows implementation.  That way we don't pay for it in either run-time cost or DLL size on Unix.  It makes sense to move it into the PAL as it's part of making this functional on Windows.